src/tuplecast.jl

```diff
@@ -0,0 +1,177 @@
+
+"""
+    tuplecast(f, args...)
```

Comment:
how do you feel about calling this:
multi_broadcast.
tuplecast isn't super clear to me,
where as multi_broadcast is an analogy to multimap. Which at least to me conveys something like this is is like doing multiple maps to get multiple outputs

Reply:
I guess "multimap" means nothing to me. What I wanted to emphasise is that this works *only* for functions returning a tuple, no other structs are accepted / iterated etc. Which is enough for what CR wants this function to do.

Reply:
I think `multicast` or `multibroadcast` is a better name implying the multiple output locations.
for `tuplecast` I intially read it as for functions that accept a tuple.
We have a nice clear error message here for if you use functions that don't return a tuple

Reply:
[a365fe3](https://github.com/JuliaDiff/ChainRules.jl/pull/644/commits/a365fe31089a41bbe2e1506e29faa583c0f72de1) makes it `unzip_broadcast`, which seems pretty explicit.

Reply:
Nice that is best